x-pack/test/fleet_api_integration/apis/agents/unenroll.ts

```diff
@@ -61,7 +60,7 @@ export default function (providerContext: FtrProviderContext) {
         },
       });
     });
-    after(async () => {
+    afterEach(async () => {
```

Comment:
Moved the cleanup to each `it`, instead of only `describe`. Many of the tests are destructive and can interfere with later tests which makes them flaky. I believe each is meant to be self-contained.

Reply:
You may still have to have a before and after  that load empty fleet server fixtures as the `setupFleetAndAgents` will probably mess the `.fleet*` indices otherwise

Reply:
I can add them both back, but do you know of any way for us to test/confirm? If they pass CI should we leave them out or would you still prefer to restore them?

Reply:
if CI pass leave them out :)

Reply:
While developing, [this test](https://github.com/jfsiii/kibana/blob/f4caf8727dd7989c31983b3f0fbf88fc3c883d4d/x-pack/test/fleet_api_integration/apis/agents/unenroll.ts#L166-L178) was failing it wouldn't remove 2 agents which were enrolled in a managed policy earlier. That's why I added the reset between tests. However, CI [kept failing in the before each for that test](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/113807/execution/node/795/log/) with a `ResponseError: invalid_alias_name_exception`. 

I reverted the changes and CI passed 🤷🏻 🎉